src/runaqa.ts

```diff
@@ -1,324 +1,475 @@
+    await exec.exec('sudo sysctl -w kern.sysv.shmall=655360')
+    await exec.exec('sudo sysctl -w kern.sysv.shmmax=125839605760')
+  } else {
+    if (fs.existsSync('/usr/bin/apt-get')) {
+      // Debian Based
+      await exec.exec('sudo apt-get update')
+      await exec.exec('sudo apt-get install ant-contrib -y')
+    } else if (fs.existsSync('/usr/bin/yum')) {
+      // RPM Based
+      await exec.exec('sudo yum update -y')
+      await exec.exec('sudo yum install p7zip -y')
+      const antContribFile = await tc.downloadTool(
+        `https://sourceforge.net/projects/ant-contrib/files/ant-contrib/ant-contrib-1.0b2/ant-contrib-1.0b2-bin.zip/download`
+      )
+      await tc.extractZip(`${antContribFile}`, `${tempDirectory}`)
+      await io.cp(
+        `${tempDirectory}/ant-contrib/lib/ant-contrib.jar`,
+        `${process.env.ANT_HOME}\\lib`
+      )
+    } else if (fs.existsSync('/sbin/apk')) {
+      // Alpine Based
+      await exec.exec('apk update')
+      await exec.exec('apk add p7zip')
+      const antContribFile = await tc.downloadTool(
+        `https://sourceforge.net/projects/ant-contrib/files/ant-contrib/ant-contrib-1.0b2/ant-contrib-1.0b2-bin.zip/download`
+      )
+      await tc.extractZip(`${antContribFile}`, `${tempDirectory}`)
+      await io.cp(
+        `${tempDirectory}/ant-contrib/lib/ant-contrib.jar`,
+        `${process.env.ANT_HOME}\\lib`
+      )
+    }
+    //environment
+    if ('RUNNER_USER' in process.env) {
+      process.env['LOGNAME'] = process.env['RUNNER_USER']
+    } else {
+      core.warning(
+        'RUNNER_USER is not the GitHub Actions environment variables shell script. Container is configured differently. Please check the updated lists of environment variables.'
+      )
+    }
+
+    if (fs.existsSync('/usr/bin/apt-get')) {
+      //disable apport
+      await exec.exec('sudo service apport stop')
+    }
+  }
+}
+
+/**
+ * set required SPEC env variable based on OS type.
+ * @return {[null]} null     [description]
+ */
+function setSpec(): void {
+  if (IS_WINDOWS) {
+    process.env['SPEC'] = 'win_x86-64_cmprssptrs'
+  } else if (IS_MACOS) {
+    process.env['SPEC'] = 'osx_x86-64_cmprssptrs'
+  } else {
+    process.env['SPEC'] = 'linux_x86-64_cmprssptrs'
+  }
+}
+
+/**
+ * Installs aqa-test repository onto the runner.
+ * @param  {string} version JDK version
+ * @param  {[string]} buildList [description]
+ * @param  {[string]} aqatestsRepo Alternative aqatestRepo
+ * @return {[null]} null
+ */
+async function getAqaTestsRepo(aqatestsRepo: string, version: string, buildList: string): Promise<void> {
+  let repoBranch = ['adoptium/aqa-tests', 'master']
+  if (aqatestsRepo.length !== 0) {
+    repoBranch = parseRepoBranch(aqatestsRepo)
+  }
+  await exec.exec(
+    `git clone --depth 1 -b ${repoBranch[1]} https://github.com/${repoBranch[0]}.git`
+  )
+  process.chdir('aqa-tests')
+  // workaround until TKG can download the artifacts required for Windows

```

Comment:
Do we have an issue number to refer to here?

Reply:
Negative. Will open one!

Reply:
Opened: https://github.com/adoptium/TKG/issues/342